src/map.cpp

```diff
@@ -2767,45 +2764,48 @@ std::pair<bool, bool> map::bash( const tripoint &p, const int str,
                 smashed_something = true;
             }
         } else {
-            furn_id furnid = furn(x, y);
+            furn_id furnid = furn(p);
             if ( furnid == f_skin_wall || furnid == f_skin_door || furnid == f_skin_door_o ||
                  furnid == f_skin_groundsheet || furnid == f_canvas_wall || furnid == f_canvas_door ||
                  furnid == f_canvas_door_o || furnid == f_groundsheet || furnid == f_fema_groundsheet) {
                 if (str >= rng(0, 6) || destroy) {
                     // Special code to collapse the tent if destroyed
-                    int tentx = -1, tenty = -1;
+                    tripoint tentp = tripoint_min;
                     // Find the center of the tent
-                    for (int i = -1; i <= 1; i++) {
-                        for (int j = -1; j <= 1; j++) {
-                            if (furn(x + i, y + j) == f_groundsheet ||
-                                furn(x + i, y + j) == f_fema_groundsheet ||
-                                furn(x + i, y + j) == f_skin_groundsheet){
-                                tentx = x + i;
-                                tenty = y + j;
+                    tripoint tmp = p;
+                    int &i = tmp.x;
+                    int &j = tmp.y;
+                    for( i = p.x - 1; i <= p.x + 1; i++ ) {
+                        for( j = p.y - 1; j <= p.y + 1; j++ ) {
+                            const auto f_at = furn( tmp );
+                            if( f_at == f_groundsheet ||
+                                f_at == f_fema_groundsheet ||
+                                f_at == f_skin_groundsheet){
+                                tentp = tmp;
                                 break;
                             }
                         }
                     }
                     // Never found tent center, bail out
-                    if (tentx == -1 && tenty == -1) {
-                        smashed_something = true;
+                    if( tentp == tripoint_min ) {
+                        std::pair<bool, bool>( true, false );
```

Comment:
This doesn't do anything, or does it? Looks like a call to the constructor of `std::pair`, but it's not used to stored or anything at all.